AMSoftware.Crm.PowerShell.Commands/Administration/GetSPDocLoc.cs

```diff
@@ -0,0 +1,65 @@
+﻿using AMSoftware.Crm.PowerShell.Common;
+using AMSoftware.Crm.PowerShell.Common.ArgumentCompleters;
+using AMSoftware.Crm.PowerShell.Common.Repositories;
+using Microsoft.Crm.Sdk.Messages;
+using Microsoft.Xrm.Sdk;
+using Microsoft.Xrm.Sdk.Query;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AMSoftware.Crm.PowerShell.Commands.Administration
+{
+    [Cmdlet(VerbsCommon.Get, "CrmSPDocLoc", HelpUri = HelpUrlConstants.GetRoleHelpUrl, SupportsPaging = true, DefaultParameterSetName = "")]
+    [OutputType(typeof(Entity))]
+    public class GetSPDocLoc :CrmCmdlet
+    {
+        [Parameter(Mandatory = true)]
+        public Guid[] Id { get; set; }
+
+        [ArgumentCompleter(typeof(EntityArgumentCompleter))]
+        [Parameter]
+        public string Entity { get; set; }
+        private ContentRepository _repository = new ContentRepository();
+        protected override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            if (String.IsNullOrEmpty(Entity))
+            {
+                foreach (var id in Id)
+                {
+                    RetrieveUrlById(id);
+                }
+
+            }
+            else
+            {
+                foreach (var id in Id)
+                {
+                    QueryByAttribute query = new QueryByAttribute("sharepointdocumentlocation");
+                    query.AddAttributeValue("regardingobjectid", id);
```

Comment:
Include Entity and EntityId in query. There won't be duplicate ID's but regarding is a lookup so includes the Entity, and is an extra check for the user.

Reply:
not exactly sure what you mean by this. Please see the latest changes and check if this is already obsolete.